CBinding/Gui/CTextEditorExtension.cs

```diff
@@ -221,382 +245,274 @@ public override bool KeyPress (KeyDescriptor descriptor)
 						
 							// Default indentation method
 							Editor.InsertAtCaret (Editor.EolMarker + indent + GetIndent(Editor, Editor.OffsetToLineNumber (line.Offset), lineCursorIndex));
-							
 							return false;
 						
 					}
 				}
 			}
-			
 			return base.KeyPress (descriptor);
 		}
 
+		/// <summary>
+		/// Determine if code completion should be invoked by pressed char's (key's) value.
+		/// </summary>
+		/// <returns><c>true</c>, if complete should be invoked, <c>false</c> otherwise.</returns>
+		/// <param name="pressed">Pressed.</param>
+		bool shouldCompleteOn(char pressed) {
+			switch (pressed) {
+			case '.':
+				return true;
+			case '(':
+				return true;
+			case ' ':
+				return true;
+			case '>':
+				if (previous == '-')
+					return true;
+				return false;
+			case ':':
+				if (previous == ':')
+					return true;
+				return false;
+			default:
+				return false;
+
+			}
+		}
+
+		/// <summary>
+		/// Handles the code completion async. Invoked automatically on every keypress.
+		/// </summary>
+		/// <returns>The code completion async.</returns>
+		/// <param name="completionContext">Completion context.</param>
+		/// <param name="completionChar">Completion char.</param>
+		/// <param name="token">Token.</param>
 		public override Task<ICompletionDataList> HandleCodeCompletionAsync (CodeCompletionContext completionContext, char completionChar, CancellationToken token = default(CancellationToken))
```

Comment:
It would be good to make this properly async but that can be saved for after the PR is merged.